### PR TITLE
Add getwatch command.

### PR DESCRIPTION
**Pull request by analytically**

### CODE REVIEW

**Comment by RevCBH** [comment]
+1

**Comment by matthis-perrin** [comment]
:+1:

File: etcdctl/command/getwatch_command.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+	index := 0
+	if c.Int("after-index") != 0 {
+		index = c.Int("after-index") + 1
+	}
+
+	// Retrieve the value from the server.
+	value, _ := client.Get(key, sorted, false)
+
+	if value == nil {
```
Comment (analytically) [inline etcdctl/command/getwatch_command.go]: Actually I should best check here for error code 100 (Key not found) and only watch then.

**Comment by bwtimm** [comment]
+1

File: etcdctl/command/getwatch_command.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		var err error
+		resp, err = client.Watch(key, uint64(index), recursive, nil, nil)
+
+		if err != nil {
+			handleError(ErrorFromEtcd, err)
+		}
+
+		if err != nil {
+			return nil, err
+		}
```
Comment (ravigadde) [inline etcdctl/command/getwatch_command.go]: value = resp.Node.Value

**Comment by analytically** [comment]
@ravigadde implemented your remarks

**Comment by ravigadde** [comment]
@analytically Still doesn't return resp.Node.Value.

Also thinking more there are a few issues/questions:

1) after-index is not specified.
If get fails and the key gets created before watch is called, watch will miss that. Need to set watch before get.
2) Is this supported on a directory? If the directory is not present, but created later, what should be returned?

I will let etcd experts comment more.

**Comment by analytically** [comment]
Regarding resp.Node.Value: cannot use resp.Node.Value (type string) as type *etcd.Response in return argument

1) watch blocks, so moving it before the get voids the intention of this command
2) on a directory it just says '/blah: is a directory' - I copied this behaviour from the watch command

I'm new to Go so I'm not sure how to fix your first point.

**Comment by ravigadde** [comment]
@analytically 
1) go Watch(...) followed by get. If get succeeds, stop the watch using stop channel that was provided to Watch. If not, process events on the watch channel.
2) Regarding the return value, I said incorrectly resp.Node.Value. I meant to say that you are not returning the resp from Watch at all. value is nil at that point and that is what gets returned.

**Comment by analytically** [comment]
@ravigadde 

1) the watch blocks, so the get never gets executed until there is a value
2) I copied this code from the watch command, and it sure is working correctly, returning actual values

**Comment by analytically** [comment]
@xiang90 any chance merging this one?

**Comment by barakmich** [comment]
I like where it's going, but I'd also like to sort of set some etcd terminology. We've been talking about it as Watch and Wait. -- Watch continually watches for updates, and Wait is, much as you've done here, returning once it's been set once. It's a watch-until-first-value. 

With some Wait cases you want the block to depend on the previous value. (In this case, value-not-found, actually) -- sometimes you just want to return if the value is anything but what you'd expect. (It's been updated before you can put the wait there)

So maybe we can shape this (which is looking pretty good) into a slightly more general `wait` subcommand. `wait` by itself merely waits unilaterally for the next value on the key and returns. `wait --wait-empty` or `wait --wait-value=foo` which does the check, and waits only if the value is what you've told it to expect (no value, or a specific value, respectively). Because you want to do it atomically and can't be guaranteed that the wait you're looking for has slipped in between two runs (`etcdctl get && etcdctl wait`) -- hence why you called it `getwatch` but I think the more general case is `wait [precondition flag]`)

What do you think?

**Comment by kelseyhightower** [comment]
@analytically Once you get a free moment can you address the concerns raised by @barakmich?

**Comment by kelseyhightower** [comment]
@analytically I'm closing this due to inactivity, please feel free to reopen this once you are ready to resume the review process.

**Pull request closed without merging**